### ISSUE TITLE
W1030 report error on Ref definition and not where the Ref does not respect a requirement

### ISSUE THREAD

**Issue by gautaz** [aws-cloudformation/cfn-lint#4446]
### CloudFormation Lint Version

cfn-lint 1.48.1

### What operating system are you using?

Linux hepao 6.18.21 #1-NixOS SMP PREEMPT_DYNAMIC Thu Apr  2 11:23:33 UTC 2026 x86_64 GNU/Linux

### Describe the bug

On the following reproduction template, cfn-lint reports this warning:

```terminal
W1030 {'Ref': 'ExampleDesiredCount'} is less than the minimum of 1 when 'Ref' is resolved
stack/test-template.yaml:5:5
```

Line 5 is in fact where the minimum value of the Ref is defined but the real culprit (the `minimum of 1`) in this template is the usage of the parameter in `DatapointsToAlarm` at line 45.

On a large template where the parameter is used multiple times, it is currently quite difficult to retrieve which Ref is causing the warning.

### Expected behavior

I would have expected a message like this one:

```terminal
W1030 {'Ref': 'ExampleDesiredCount'} is less than the minimum of 1 when 'Ref' is resolved
Ref minimum at stack/test-template.yaml:5:5 is lower than the minimum needed at stack/test-template.yaml:45:26
```

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Parameters:
  ExampleDesiredCount:
    Default: 1
    MinValue: 0
    Type: Number
Resources:
  ExampleService:
    Properties:
      AvailabilityZoneRebalancing: ENABLED
      DesiredCount: !Ref 'ExampleDesiredCount'
      LaunchType: FARGATE
      NetworkConfiguration:
        AwsvpcConfiguration:
          SecurityGroups:
            - !GetAtt 'ExampleServiceToAwsServicesSecurityGroup.GroupId'
      TaskDefinition: !Ref 'ExampleTaskDefinition'
    Type: AWS::ECS::Service
  ExampleServiceToAwsServicesSecurityGroup:
    Properties:
      GroupDescription: AWS services access
      SecurityGroupEgress:
        - CidrIp: '0.0.0.0/0'
          FromPort: 443
          IpProtocol: TCP
          ToPort: 443
      VpcId: !Ref 'VPC'
    Type: AWS::EC2::SecurityGroup
  ExampleTaskDefinition:
    Properties:
      ContainerDefinitions:
        - Image: example-image:1.2.3
          Name: ExampleContainer
          PortMappings:
            - ContainerPort: 8080
      Cpu: '256'
      Memory: '512'
      NetworkMode: awsvpc
      RequiresCompatibilities:
        - FARGATE
    Type: AWS::ECS::TaskDefinition
  ExampleServiceMissingTasksOverTimeAlarm:
    Properties:
      ComparisonOperator: GreaterThanThreshold
      DatapointsToAlarm: !Ref 'ExampleDesiredCount'
      EvaluationPeriods: 15
      Metrics:
        - Id: desired
          MetricStat:
            Metric:
              Dimensions:
                - Name: ServiceName
                  Value: !GetAtt 'ExampleService.Name'
              MetricName: DesiredTaskCount
              Namespace: ECS/ContainerInsights
            Period: 120
            Stat: Average
          ReturnData: false
        - Id: running
          MetricStat:
            Metric:
              Dimensions:
                - Name: ServiceName
                  Value: !GetAtt 'ExampleService.Name'
              MetricName: RunningTaskCount
              Namespace: ECS/ContainerInsights
            Period: 120
            Stat: Average
          ReturnData: false
        - Expression: desired - running
          Id: missing
          Label: Missing task count
          ReturnData: true
      Threshold: 0.5
      TreatMissingData: breaching
    Type: AWS::CloudWatch::Alarm
  VPC:
    Properties:
      CidrBlock: 10.0.0.0/16
      EnableDnsHostnames: true
      EnableDnsSupport: true
    Type: AWS::EC2::VPC
```
